components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/PublicClientAuthenticator.java

```diff
@@ -80,6 +81,25 @@ public boolean authenticateClient(HttpServletRequest request, Map<String, List>
     public boolean canAuthenticate(HttpServletRequest request, Map<String, List> bodyParams, OAuthClientAuthnContext
             context) {
 
+        List publicClientSupportedGrantTypes = OAuthServerConfiguration.getInstance().
+                getPublicClientSupportedGrantTypesList();
+        List grantTypes = bodyParams.get("grant_type");
```

Comment:
"grant_type" need to be made a constant or reuse existing constant

Reply:
Fixed